trending_data/src/main/java/com/google/sps/servlets/TwitterServlet.java

```diff
@@ -0,0 +1,64 @@
+import com.google.gson.Gson;
+import com.google.sps.data.TrendingTopic;
+import com.google.sps.data.TwitterResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/twitter")
+public class TwitterServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json; charset=UTF-8");
+    String woeid = request.getParameter("woeid");
+
+    Collection<TrendingTopic> trendingTopics = getTrendingTopics(woeid);
+    response.getWriter().println(new Gson().toJson(trendingTopics));
+  }
+
+  /** returns a Collection of all the TrendingTopics returned by the Twitter API for this WOEID */
+  private Collection<TrendingTopic> getTrendingTopics(String woeid) {
+    // get all twitter responses
+    TwitterResponse[] twitterResponses = getTwitterResponses(woeid);
+
+    Collection<TrendingTopic> trendingTopics = new ArrayList<TrendingTopic>();
+
+    // for each twitter response add all trending topics in it to the
+    // collections of trending topics
+    for (TwitterResponse twitterResponse : twitterResponses) {
+      Collections.addAll(trendingTopics, twitterResponse.getTrends());
+    }
+    return trendingTopics;
+  }
+  /**
+   * returns an array of TwitterResponses where each twitter response contains an array of
+   * TrendingTopics for this WOEID
+   */
+  private TwitterResponse[] getTwitterResponses(String woeid) {
```

Comment:
Same comment about arrays vs collections here - you should return an immutable list of TwitterResponse instead of an array.